### PR TITLE
Add protoc-gen-openapi v0.7.0

### LINKED ISSUE #821: Plugin request for Buf Schema Registry [protoc-gen-openapi]
## Mandatory

**Where is the source code for the plugin?**

Example, the source code for the `protoc-gen-connect-go` plugin is found here:

https://github.com/google/gnostic/tree/main/cmd/protoc-gen-openapi

## Optional

**Does the plugin have a valid semver version?**

Yes

What is the latest version, and where did you get this value from?

cmd/protoc-json-openapi/v0.6.5

https://github.com/google/gnostic/releases/tag/cmd%2Fprotoc-json-openapi%2Fv0.6.5

**Does the plugin have a dependency on another plugin?**

Not to my knowledge.

**Does the output of this plugin depend on any other external libraries?**

Not to my knowledge.

@timburks please correct if necessary

### PR DESCRIPTION
**Pull request by mfridman**
Fix #821

### CODE REVIEW

File: plugins/community/google-gnostic-openapi/v0.7.0/buf.plugin.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+version: v1
+name: buf.build/community/google-gnostic-openapi
```
Comment (mfridman) [inline plugins/community/google-gnostic-openapi/v0.7.0/buf.plugin.yaml:2]:
```suggestion
name: buf.build/community/google-openapi
```

This alternative drops the "gnostic" from the name, which would be more in line with our existing nomenclature conventions of: `{owner}-{name}`

Where the repository is the upstream GitHub owner and the name is usually the name of the plugin minus `protoc-gen-` prefix.

However, for this particular one having the name "gnostic" in the name makes the most sense given the project's background. For some context:

- https://github.com/grpc-ecosystem/grpc-gateway/issues/441
- https://github.com/google/gnostic/issues/340

Although this is parked under github.com/gnostic organization,

**Comment by mfridman** [comment]
If we have some cycles, we should consider adding a sample image that imports:

```
import "google/api/annotations.proto";
import "openapiv3/annotations.proto";
```

That way we exercise bits of the plugin that actually use the annotations.

File: plugins/community/google-gnostic-openapi/v0.7.0/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# syntax=docker/dockerfile:1.6
+FROM golang:1.21.1-bullseye AS build
```
Comment (pkwarren) [inline plugins/community/google-gnostic-openapi/v0.7.0/Dockerfile:2]:
```suggestion
FROM golang:1.21.2-bullseye AS build
```

File: plugins/community/google-gnostic-openapi/v0.7.0/buf.plugin.yaml (hunk 1 above)
Comment (pkwarren) [reply plugins/community/google-gnostic-openapi/v0.7.0/buf.plugin.yaml:2]: I prefer having the `gnostic` in the name too.

**Comment by pkwarren** [comment]
> If we have some cycles, we should consider adding a sample image that imports:
> 
> ```
> import "google/api/annotations.proto";
> import "openapiv3/annotations.proto";
> ```
> 
> That way we exercise bits of the plugin that actually use the annotations.

We did something similar to verify the grpc-gateway code. I think it would be good to do the same here but at least there is an openapi.yaml file generated for our existing inputs (even if it is mostly empty).

**Review by pkwarren — APPROVED** [review]

**Pull request merged**